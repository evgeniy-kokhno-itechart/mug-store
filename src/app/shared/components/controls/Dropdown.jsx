import React from 'react';
import { PropTypes } from 'prop-types';
import FormGroup from '../markup/FormGroup';
import ControlBase from './ControlBase';

class Dropdown extends ControlBase {
  render() {
    const {
      name, label, error, value, options, defaultText, labelClasses, selectClasses, wrapperClasses,
    } = this.props;
    return (
      <FormGroup
        name={name}
        label={label}
        error={error}
        groupClasses={`form-group flex-nowrap ${wrapperClasses}`}
        labelClasses={`mr-2 text-nowrap ${labelClasses}`}
      >
        <select name={name} id={name} value={value} onChange={this.handleChange} className={selectClasses}>
          {defaultText && <option>{defaultText}</option>}
          {options.map(o => (
            <option key={o.id} value={o.id}>
              {o.name}
            </option>
          ))}
        </select>
      </FormGroup>
    );
  }
}

Dropdown.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  error: PropTypes.string,
  value: PropTypes.string,
  options: PropTypes.arrayOf(PropTypes.shape({ id: PropTypes.string, name: PropTypes.string })).isRequired,
  defaultText: PropTypes.string,
  labelClasses: PropTypes.string,
  selectClasses: PropTypes.string,
  wrapperClasses: PropTypes.string,
};

Dropdown.defaultProps = {
  error: '',
  defaultText: '',
  labelClasses: '',
  selectClasses: '',
  wrapperClasses: '',
  value: '',
};

export default Dropdown;
