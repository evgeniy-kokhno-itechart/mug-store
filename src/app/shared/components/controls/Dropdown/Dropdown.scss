.dropdown {

    &__label {
        margin: 0 0.5rem 0 0.5rem;
        white-space: nowrap;
    }

    &__selector {
        font: inherit;
    }
}