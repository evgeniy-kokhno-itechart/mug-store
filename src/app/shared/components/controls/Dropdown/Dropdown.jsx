import React from 'react';
import { PropTypes } from 'prop-types';
import FormGroup from '../../markup/FormGroup/FormGroup';
import ControlBase from '../ControlBase';
import './Dropdown.scss';

class Dropdown extends ControlBase {
  render() {
    const {
      name, label, error, value, options, defaultText, labelClasses, selectClasses, wrapperClasses,
    } = this.props;
    return (
      <FormGroup
        name={name}
        label={label}
        error={error}
        groupClasses={`dropdown ${wrapperClasses}`}
        labelClasses={`dropdown__label ${labelClasses}`}
      >
        <select
          name={name}
          id={name}
          value={value}
          onChange={this.handleChange}
          className={`dropdown__selector focusable ${selectClasses}`}
        >
          {defaultText && <option>{defaultText}</option>}
          {options.map(o => (
            <option key={o.id} value={o.id}>
              {o.name}
            </option>
          ))}
        </select>
      </FormGroup>
    );
  }
}

Dropdown.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  error: PropTypes.string,
  value: PropTypes.string,
  options: PropTypes.arrayOf(PropTypes.shape({ id: PropTypes.string, name: PropTypes.string })).isRequired,
  defaultText: PropTypes.string,
  labelClasses: PropTypes.string,
  selectClasses: PropTypes.string,
  wrapperClasses: PropTypes.string,
};

Dropdown.defaultProps = {
  error: '',
  defaultText: '',
  labelClasses: '',
  selectClasses: '',
  wrapperClasses: '',
  value: '',
};

export default Dropdown;
