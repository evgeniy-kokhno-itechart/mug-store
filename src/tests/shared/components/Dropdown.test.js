/* eslint-disable no-undef */
import React from 'react';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Dropdown from '../../../app/shared/components/controls/Dropdown';

configure({ adapter: new Adapter() });

describe('<Dropdown />', () => {
  let DropdownWrapper;

  beforeEach(() => {
    DropdownWrapper = shallow(
      <Dropdown
        name="name"
        label="label"
        options={[{ id: '1', name: 'test name 1' }, { id: '2', name: 'test name 2' }]}
      />,
    );
  });

  it('renders properly with default props', () => {
    expect(DropdownWrapper).toMatchSnapshot();
  });

  it('renders properly with error', () => {
    DropdownWrapper.setProps({ error: 'test error' });
    expect(DropdownWrapper).toMatchSnapshot();
  });

  it('renders properly with value provided', () => {
    DropdownWrapper.setProps({ value: '2' });
    expect(DropdownWrapper).toMatchSnapshot();
  });

  it('renders properly with defaultText', () => {
    DropdownWrapper.setProps({ defaultText: 'test default text' });
    expect(DropdownWrapper).toMatchSnapshot();
  });

  it('renders properly with labelClasses', () => {
    DropdownWrapper.setProps({ labelClasses: 'testlabelclass' });
    expect(DropdownWrapper).toMatchSnapshot();
  });

  it('renders properly with selectClasses', () => {
    DropdownWrapper.setProps({ selectClasses: 'testselectclass' });
    expect(DropdownWrapper).toMatchSnapshot();
  });

  it('renders properly with wrapperClasses', () => {
    DropdownWrapper.setProps({ wrapperClasses: 'testwrapperclass' });
    expect(DropdownWrapper).toMatchSnapshot();
  });
});
