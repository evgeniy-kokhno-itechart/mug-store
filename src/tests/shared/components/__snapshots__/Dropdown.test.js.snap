// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> renders properly with default props 1`] = `
<FormGroup
  error=""
  groupClasses="dropdown "
  label="label"
  labelClasses="dropdown__label "
  name="name"
>
  <select
    className="dropdown__selector focusable "
    id="name"
    name="name"
    onChange={[Function]}
    value=""
  >
    <option
      key="1"
      value="1"
    >
      test name 1
    </option>
    <option
      key="2"
      value="2"
    >
      test name 2
    </option>
  </select>
</FormGroup>
`;

exports[`<Dropdown /> renders properly with defaultText 1`] = `
<FormGroup
  error=""
  groupClasses="dropdown "
  label="label"
  labelClasses="dropdown__label "
  name="name"
>
  <select
    className="dropdown__selector focusable "
    id="name"
    name="name"
    onChange={[Function]}
    value=""
  >
    <option>
      test default text
    </option>
    <option
      key="1"
      value="1"
    >
      test name 1
    </option>
    <option
      key="2"
      value="2"
    >
      test name 2
    </option>
  </select>
</FormGroup>
`;

exports[`<Dropdown /> renders properly with error 1`] = `
<FormGroup
  error="test error"
  groupClasses="dropdown "
  label="label"
  labelClasses="dropdown__label "
  name="name"
>
  <select
    className="dropdown__selector focusable "
    id="name"
    name="name"
    onChange={[Function]}
    value=""
  >
    <option
      key="1"
      value="1"
    >
      test name 1
    </option>
    <option
      key="2"
      value="2"
    >
      test name 2
    </option>
  </select>
</FormGroup>
`;

exports[`<Dropdown /> renders properly with labelClasses 1`] = `
<FormGroup
  error=""
  groupClasses="dropdown "
  label="label"
  labelClasses="dropdown__label testlabelclass"
  name="name"
>
  <select
    className="dropdown__selector focusable "
    id="name"
    name="name"
    onChange={[Function]}
    value=""
  >
    <option
      key="1"
      value="1"
    >
      test name 1
    </option>
    <option
      key="2"
      value="2"
    >
      test name 2
    </option>
  </select>
</FormGroup>
`;

exports[`<Dropdown /> renders properly with selectClasses 1`] = `
<FormGroup
  error=""
  groupClasses="dropdown "
  label="label"
  labelClasses="dropdown__label "
  name="name"
>
  <select
    className="dropdown__selector focusable testselectclass"
    id="name"
    name="name"
    onChange={[Function]}
    value=""
  >
    <option
      key="1"
      value="1"
    >
      test name 1
    </option>
    <option
      key="2"
      value="2"
    >
      test name 2
    </option>
  </select>
</FormGroup>
`;

exports[`<Dropdown /> renders properly with value provided 1`] = `
<FormGroup
  error=""
  groupClasses="dropdown "
  label="label"
  labelClasses="dropdown__label "
  name="name"
>
  <select
    className="dropdown__selector focusable "
    id="name"
    name="name"
    onChange={[Function]}
    value="2"
  >
    <option
      key="1"
      value="1"
    >
      test name 1
    </option>
    <option
      key="2"
      value="2"
    >
      test name 2
    </option>
  </select>
</FormGroup>
`;

exports[`<Dropdown /> renders properly with wrapperClasses 1`] = `
<FormGroup
  error=""
  groupClasses="dropdown testwrapperclass"
  label="label"
  labelClasses="dropdown__label "
  name="name"
>
  <select
    className="dropdown__selector focusable "
    id="name"
    name="name"
    onChange={[Function]}
    value=""
  >
    <option
      key="1"
      value="1"
    >
      test name 1
    </option>
    <option
      key="2"
      value="2"
    >
      test name 2
    </option>
  </select>
</FormGroup>
`;
